### PR TITLE
introduce aligned_stored_size to flush write cache

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
`STORE_META_OVERHEAD` is too large. The result is we way over-allocate append vecs when we flush the write cache. This causes us to need to shrink later unnecessarily soon.

#### Summary of Changes
Calculate the required file size accurately.

Fixes #

### CODE REVIEW

**Comment by carllin** [comment]
Good catch! Would be good to have a test of some sort that verifies this length

**Comment by jeffwashington** [comment]
> Good catch! Would be good to have a test of some sort that verifies this length

yep, I'm still testing it. Isn't quite right yet.

File: runtime/src/append_vec.rs [hunk 1]
```diff
@@ -41,6 +41,17 @@ macro_rules! u64_align {
...
 }
 
+/// size of the fixed sized fields in an append vec
+/// we need to add data len and align it to get the actual stored size
+const STORE_META_OVERHEAD: usize = 136;
+
+/// Returns the size this item will take to store plus possible alignment padding bytes before the next entry.
+/// fixed-size portion of per-account data written
+/// plus 'data_len', aligned to next boundary
+pub fn aligned_stored_size(data_len: usize) -> usize {
```
[the remark this answers is not in the document]
Comment (jeffwashington) [reply runtime/src/append_vec.rs:58]: source data is data.len(). len() is well defined in rust. 'size' is more of a concept of how many bytes and seems different than len. Even the type `usize` is a size. Now, `len()` returns usize. I think I'm happy with it having these names. What do you suggest?

**Comment by jeffwashington** [comment]
> Good catch! Would be good to have a test of some sort that verifies this length

this is tested. I tagged you in this pr.

**Comment by jeffwashington** [comment]
> Oh, one other thing! Can you confirm that with this change we can still open up snapshots created before this change?
> 
> I think it should be fine, since `get_account()` gets the `StoredMeta`, `AccountMeta`, and `Hash` explicitly, but want to confirm.

yes, ledger-tool verify runs on mnb snapshot. validator started and is making roots on mnb

**Pull request merged**